homeassistant/components/leviosa_shades/cover.py

```diff
@@ -0,0 +1,180 @@
+"""The Leviosa Shades Zone base entity."""
+import logging
+
+from leviosapy import LeviosaShadeGroup as tShadeGroup, LeviosaZoneHub as tZoneHub
+import voluptuous as vol
+
+from homeassistant.components import cover
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    BLIND_GROUPS,
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+    SERVICE_NEXT_DOWN_POS,
+    SERVICE_NEXT_UP_POS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Estimated time it takes to complete a transition
+# from one state to another
+TRANSITION_COMPLETE_DURATION = 30
+PARALLEL_UPDATES = 1
+
+COVER_NEXT_POS_SCHEMA = vol.Schema({vol.Optional(ATTR_ENTITY_ID): cv.entity_ids})
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Leviosa shade groups."""
+    _LOGGER.debug(
+        "Setting up %s[%s]: %s",
+        entry.domain,
+        entry.title,
+        entry.entry_id,
+    )
+    hub_name = entry.title
+    hub_mac = entry.data["device_mac"]
+    hub_ip = entry.data["host"]
+    blind_groups = entry.data[BLIND_GROUPS]
+    _LOGGER.debug("Groups to create: %s", blind_groups)
+    hub = tZoneHub(
+        hub_ip=hub_ip, hub_name=hub_name, websession=async_get_clientsession(hass)
+    )
+    await hub.getHubInfo()  # Check all is good
+    _LOGGER.debug("Hub object created, FW: %s", hub.fwVer)
+    entities = []
+    for blind_group in blind_groups:
+        _LOGGER.debug("Adding blind_group: %s", blind_group)
+        new_group_obj = hub.AddGroup(blind_group)
+        entities.append(
+            LeviosaBlindGroup(
+                hass, hub_mac + "-" + str(new_group_obj.number), new_group_obj
+            )
+        )
+    async_add_entities(entities)
+
+    _LOGGER.debug("Setting up Leviosa shade group services")
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        SERVICE_NEXT_DOWN_POS,
+        COVER_NEXT_POS_SCHEMA,
+        "next_down_pos",
+    )
+    platform.async_register_entity_service(
+        SERVICE_NEXT_UP_POS,
+        COVER_NEXT_POS_SCHEMA,
+        "next_up_pos",
+    )
+
+
+class LeviosaBlindGroup(cover.CoverEntity):
+    """Represents a Leviosa shade group entity."""
+
+    def __init__(
+        self, hass: HomeAssistant, blind_group_id, blind_group_obj: tShadeGroup
+    ) -> None:
+        """Initialize the shade group."""
+        self._blind_group_id = blind_group_id
+        self._blind_group_obj = blind_group_obj
+        self._hass = hass
+        _LOGGER.debug(
+            "Creating cover.%s, UID: %s",
+            self._blind_group_obj.name,
+            self._blind_group_id,
+        )
+
+    @property
```

Comment:
sorry I don't get you.
Are you referring to programming style issues? Or do we have to name private attribute definitions in subclasses starting with _attr_?

Reply:
Sorry, I mean that instead of defining properties like:
```python
    @property
    def unique_id(self):
        """Return a unique ID for this device."""

        return self._blind_group_id
```

You can instead use in the constructor:
```python
self._attr_unique_id = blind_group_id
```

I think most of the properties have a corresponding `_attr_xxx` attribute, maybe should take a look at:
[homeassistant/components/cover/__init__.py](https://github.com/home-assistant/core/blob/6d1876394ef9eceee38c2cf9082f45d37218be07/homeassistant/components/cover/__init__.py#L222C14-L222C14)
and
https://github.com/home-assistant/core/blob/6d1876394ef9eceee38c2cf9082f45d37218be07/homeassistant/helpers/entity.py#L240

It is just a programming style issue but having a briefer code would help readability I believe.

Reply:
Your suggestions are very constructive, but I'd like to focus my time on all the code style issues (not just this one project), so would it be possible to allow this pending style change to go through first, and update it later, I don't want to spend my energy on just this one code style change.